### PR TITLE
fix: rename v2 to v1 and v1 to v0

### LINKED ISSUE #2030: rename v2 to v1
> I think I've said this before, but I call them V0 and V1 recipes. The original recipe format doesn't have a schema_version field and the new format has schema_version set to 1. I'm sure a lot of people will get confused if V2 implies a value of 1.

As @schuylermartin45 points out, v2 is really v1. We should rename again in the code. 

Cc @wolf @isuruf @nichmor

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry
* [x] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

closes #2030

### CODE REVIEW

File: news/rename-v2-v1.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
+**Added:**
+
+* <news item>
+
+**Changed:**
+
+* The conda recipe version was renamed from ``{1, 2}`` to ``{0, 1}``. (#)
```
Comment (beckermr) [inline news/rename-v2-v1.rst:7]:
```suggestion
* The conda recipe version was renamed from ``{1, 2}`` to ``{0, 1}``. (#2031)
```

File: tests/test_lint_recipe.py [hunk 2]
```diff
@@ -2618,7 +2618,7 @@ def test_pin_compatible_in_run_exports(recipe_version: int):
 
 @pytest.mark.parametrize("recipe_version", [1, 2])
```
Comment (beckermr) [inline tests/test_lint_recipe.py:2619]: This needs fixing.

**Review by beckermr — COMMENTED** [review]
I missed a few items.

File: tests/test_configure_feedstock.py [hunk 3]
```diff
@@ -224,12 +224,11 @@ def test_stdlib_on_azure(stdlib_recipe, jinja_env, request):
...
-    configure_feedstock.render_azure(
-        jinja_env=jinja_env,
-        forge_config=stdlib_recipe.config,
-        forge_dir=stdlib_recipe.recipe,
-    )
+        configure_feedstock.render_azure(
+            jinja_env=jinja_env,
+            forge_config=stdlib_recipe.config,
+            forge_dir=stdlib_recipe.recipe,
+        )
```
Comment (beckermr) [inline tests/test_configure_feedstock.py:231]:
```suggestion
    configure_feedstock.render_azure(
        jinja_env=jinja_env,
        forge_config=stdlib_recipe.config,
        forge_dir=stdlib_recipe.recipe,
    )
```

File: tests/test_configure_feedstock.py [hunk 4]
```diff
@@ -217,11 +217,18 @@ def test_py_matrix_on_azure(py_recipe, jinja_env):
...
 def test_stdlib_on_azure(stdlib_recipe, jinja_env, request):
-    configure_feedstock.render_azure(
-        jinja_env=jinja_env,
-        forge_config=stdlib_recipe.config,
-        forge_dir=stdlib_recipe.recipe,
-    )
+    conda_build_param = request.node.callspec.params["config_yaml"]
+    if conda_build_param == "rattler-build":
+        # stdlib is not yet implemented in rattler-build
+        # https://github.com/prefix-dev/rattler-build/issues/239
```
Comment (wolfv) [inline tests/test_configure_feedstock.py]: This is wrong, it's implemented :)

File: tests/test_configure_feedstock.py [hunk 5]
```diff
@@ -224,7 +224,6 @@ def test_stdlib_on_azure(stdlib_recipe, jinja_env, request):
         pytest.skip(
             "skipping test for rattler-build usecase as we currently we don't have stdlib"
         )
-
     configure_feedstock.render_azure(
```
Comment (beckermr) [inline tests/test_configure_feedstock.py:227]:
```suggestion

    configure_feedstock.render_azure(
```

**Comment by wolfv** [comment]
Yeeeeeeeees! Nice work on getting it green! :)

**Review by wolfv — APPROVED** [review]
Thanks Matt!

**Pull request merged**